build_runner/lib/src/entrypoint/daemon.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:build_daemon/constants.dart';
+import 'package:build_daemon/src/daemon.dart';
+
+import '../daemon/daemon_builder.dart';
+import 'base_command.dart';
+
+/// A command that starts the Build Daemon.
+class DaemonCommand extends BuildRunnerCommand {
+  @override
+  String get description => 'Starts the build daemon.';
+
+  @override
+  bool get hidden => true;
+
+  @override
+  String get name => 'daemon';
+
+  @override
+  Future<int> run() async {
+    var workingDirectory = Directory.current.path;
+    var options = readOptions();
+    var daemon = Daemon(workingDirectory);
+    var requestedOptions = argResults.arguments.toSet();
+    if (!daemon.tryGetLock()) {
+      var runningOptions = currentOptions(workingDirectory);
+      var version = runningVersion(workingDirectory);
+      if (version != currentVersion) {
+        print('Running Version: $version');
```

Comment:
Normally we expect output to go through the logger - but that could impact some of the things that need to get parsed precisely by process which started the daemon...

I _think_ it's OK to use print here - @jakemac53 do you have any concerns?